cogs/stockmarket/market.py

```diff
@@ -0,0 +1,568 @@
+    [p]market holdings - Displays the user's holdings
+    [p]market buy <ticker> - Buys a stock #Note: feels clunky, might as well add an <amount> param
+    [p]market sell <ticker> - Sells a stock #Note: ^^^
+    [p]market info <ticker> - Displays information about a stock WIP
+    [p]market chart <ticker> - Displays a chart of a stock
+    """
+    start = 0
+    end = 0
+
+    member_data = {
+        "transactions": [],
+        "holdings": []
+    }
+
+    def __init__(self, bot):
+        self.bot = bot
+        self.config = Config.get_conf(self, identifier=835888251, force_registration=True)
+        self.config.register_member(**self.member_data)
+        self.enable_bg_loop()
+
+    def enable_bg_loop(self):
+        """Enables the background loop"""
+        self.bg_loop_task = self.bot.loop.create_task(self.bg_loop())
+
+    def cog_unload(self):
+        """Clean up when cog shuts down."""
+        if self.bg_loop_task:
+            self.bg_loop_task.cancel()
+
+    async def bg_loop(self):
+        """Background loop to check for dividends and pay it out when it comes time"""
+        await self.bot.wait_until_ready()
+        while True:
+            await self.check_events()
+            await asyncio.sleep(86400)
+
+    async def check_events(self):
+        """Checks if any events are due and pays them out"""
+
+        # Surely there's something better than this
+        all_member_data = await self.config.all_members(guild=None)
+        for singular_guild_id in all_member_data:
+            for singular_member_id in all_member_data[singular_guild_id]:
+                member_dict = all_member_data[singular_guild_id][singular_member_id]
+                for holding in member_dict['holdings']:
+
+                    ticker = holding['ticker']
+                    actions = yf.Ticker(ticker).actions
+
+                    date = str(actions.index[-1])[:-9]
+                    most_recent_unix = time.mktime(datetime.datetime.strptime(date, "%Y-%m-%d").timetuple())
+
+                    current_unix = time.time()
+
+                    # claimable events
+                    if current_unix > most_recent_unix > holding['initial DOP']:
+
+                        # pay out the dividend and/or split
+                        dividend_due_per_share = actions.iloc[-1][0]
+                        split_due_per_share = actions.iloc[-1][1]
+
+                        # order doesn't matter since companies always try to avoid paying out dividends/splits at the same time
+                        # pay out the dividend
+                        shares_owned = holding['amount']
+                        member_object = None  # expected to change
+
+                        # pay dividend
+                        if dividend_due_per_share > 0:
+                            guild_object = self.bot.get_guild(singular_guild_id)
+                            member_object = await guild_object.query_members(user_ids=singular_member_id)

```

Comment:
1. This is a list of member objects, can you please rename this variable to describe it as such?
2. It doesn't look like we're checking for member existence (i.e. member configured a few things beforehand but then leaves the guild). Please add a check to ensure the member object actually exists, and continue to the next member if it doesn't.